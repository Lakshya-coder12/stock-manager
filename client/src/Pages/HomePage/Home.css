@media (max-width: 480px) {
  .main {
    width: 90vw;
    height: 100vh;
    margin: 0 auto;
    margin-top: 100px;
    padding: 5px;
  }
  .logout {
    position: absolute;
    top: 5px;
    right: 5px;
    display: block;
    margin-left: auto;
  }
}

@media (min-width: 480px) and (max-width: 960px) {
  .main {
    width: 50vw;
    height: 100vh;
    margin: 0 auto;
    margin-top: 100px;
    padding: 5px;
  }
  .logout {
    position: absolute;
    top: 10px;
    right: 10px;
    display: block;
    margin-left: auto;
  }
}

@media (min-width: 960px) {
  .main {
    width: 35vw;
    height: 100vh;
    margin: 0 auto;
    margin-top: 100px;
    padding: 5px;
  }
  .logout {
    position: absolute;
    top: 10px;
    right: 20px;
    display: block;
    margin-left: auto;
  }
}

.title {
  color: #1976c2;
  font-size: x-large;
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen,
    Ubuntu, Cantarell, "Open Sans", "Helvetica Neue", sans-serif;
  text-align: center;
  margin: 10px 0px;
}

.btn1 {
  margin-right: 10px;
  display: flex;
  justify-content: flex-end;
}

.stock-list {
  padding-bottom: 15px;
}
