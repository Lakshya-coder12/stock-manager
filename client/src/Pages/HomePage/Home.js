import React, { Component } from "react";
import { Link } from "react-router-dom";
import Button from "../../Components/Button/Button";
import StockCardList from "../../Components/StockCardList/StockCardList";
import "./Home.css";
// import { connect } from "react-redux";
// import { fetchAllItems } from "../../redux/actions/ItemActions";
import { PropTypes } from "prop-types";
import jwt_decode from "jwt-decode";

class Home extends Component {
  constructor(props) {
    super(props);
    this.state = {
      username: "",
    };
  }
  componentDidMount() {
    const token = localStorage.getItem("jwt");
    if (!token) {
      this.props.history.push("/login");
    } else {
      const user = jwt_decode(token);
      const username = user.user.name;
      this.setState({
        username: username,
      });
      this.props.fetchAllItems();
    }
  }

  handleLogout = () => {
    this.props.logout(this.props.history);
    this.props.resetItems();
  };

  render() {
    const allItems = this.props.allItems;
    if (allItems.loading) {
      return <div>Loading........</div>;
    } else {
      return (
        <>
          <h1 className="title">STOCK MANAGER</h1>
          <div className="main">
            <div className="btn1">
              <Link to="/new">
                <Button>New Stock</Button>
              </Link>
            </div>
            <div className="stock-list">
              <StockCardList items={allItems.items} />
            </div>
          </div>
          <div className="logout">
            <span className="username">{this.state.username}</span>
            <Button onClick={this.handleLogout}>Logout</Button>
          </div>
        </>
      );
    }
  }
}
Home.propTypes = {
  fetchAllItems: PropTypes.func,
  logout: PropTypes.func,
};
// const mapStateToProps = (state) => {
//   return {
//     allItems: state.allItems,
//   };
// };

// const mapDispatchToProps = (dispatch) => {
//   return {
//     fetchAllItems: () => dispatch(fetchAllItems()),
//   };
// };

// export default connect(mapStateToProps, mapDispatchToProps)(Home);
export default Home;
