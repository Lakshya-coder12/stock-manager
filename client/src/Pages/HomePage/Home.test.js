import { configure, shallow } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import Home from "./Home";

configure({ adapter: new Adapter() });

describe("Testing Home page", () => {
  it("matches snapshot", () => {
    const itemsMock = jest.fn();
    const pushMock = jest.fn();
    const fetchAllItemsMock = jest.fn();
    const wrapper = shallow(
      <Home
        fetchAllItems={fetchAllItemsMock}
        history={{ push: pushMock }}
        allItems={itemsMock}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
