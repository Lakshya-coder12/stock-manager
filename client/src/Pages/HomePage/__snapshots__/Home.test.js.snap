// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Home page matches snapshot 1`] = `
<Fragment>
  <h1
    className="title"
  >
    STOCK MANAGER
  </h1>
  <div
    className="main"
  >
    <div
      className="btn1"
    >
      <Link
        to="/new"
      >
        <Button>
          New Stock
        </Button>
      </Link>
    </div>
    <div
      className="stock-list"
    >
      <StockCardList />
    </div>
  </div>
  <div
    className="logout"
  >
    <span
      className="username"
    />
    <Button
      onClick={[Function]}
    >
      Logout
    </Button>
  </div>
</Fragment>
`;
